src/schemas/text-based-fields.ts

```diff
@@ -31,4 +31,11 @@ export const EMAIL_SCHEMA = yup.object<IBaseTextBasedFieldAttributes>().shape({
                 .required("Error message required."),
         })
     ),
+    conditionalRendering: yup.array().of(
+        yup.object().shape({
+            fieldKey: yup.string().required("Reference required."),
```

Comment:
understand this is causing the validation to fail if a field does not have an id, I will accept it for now because we should be looking to auto generate the id on adding an element